### PR TITLE
Upgrade to Calcite 1.32.0

### PR DESCRIPTION
**Pull request by liam-verta**
Fixes #XXXX.

### Description

This superceeds #13103 and #13149.   Calcite upgrade is quite a bit more work than just a "bump". 

I have gotten this far but the `sql` build is now failing with the following output: 
```
[INFO] --- javacc-maven-plugin:2.4:javacc (javacc) @ druid-sql ---
Java Compiler Compiler Version 4.0 (Parser Generator)
(type "javacc" with no arguments for help)
Reading from file /Users/liam/code/apache/druid/sql/target/generated-sources/annotations/javacc/Parser.jj . . .
Note: UNICODE_INPUT option is specified. Please make sure you create the parser/lexer using a Reader with the correct character encoding.
Warning: Lookahead adequacy checking not being performed since option LOOKAHEAD is more than 1.  Set option FORCE_LA_CHECK to true to force checking.
Error: Line 1228, Column 7: Non-terminal OrderItem has not been defined.
Error: Line 1233, Column 30: Non-terminal OrderItem has not been defined.
Error: Line 1453, Column 22: Non-terminal WhereOpt has not been defined.
Detected 3 errors and 1 warnings.
```

@adarshsanjeev  @LakshSingla   or anyone else with expertise in this area, help would be appreciated.

#### Added a forbidden-apis entry ...

<hr>

##### Key changed/added classes in this PR
 * `MyFoo`
 * `OurBar`
 * `TheirBaz`

<hr>

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: sql/pom.xml [hunk 1]
```diff
@@ -339,6 +338,47 @@
...
+              </exclusion>
+            </exclusions>
+          </dependency>
+          <dependency>
+            <groupId>org.freemarker</groupId>
+            <artifactId>freemarker</artifactId>
+            <version>2.3.31</version>
+          </dependency>
+        </dependencies>
+      </plugin> -->
```
Comment (liam-verta) [inline sql/pom.xml:381]:
```suggestion                  
```

**Comment by abhishekagarwal87** [comment]
We cannot upgrade calcite without shading their guava dependencies. We can't upgrade our own guava because we have to support Hadoop 2. Even after shading, there is going to be a fair amount of work involved - https://github.com/apache/druid/pull/12258

**Comment by FrankChen021** [comment]
@abhishekagarwal87 Maybe we can disable the dependency upgrade PR raised by the dependabot?

**Comment by abhishekagarwal87** [comment]
@FrankChen021 - Thanks for the tip. I will do that.

**Comment by liam-verta** [comment]
Closing duplicated effort.

**Pull request closed without merging**